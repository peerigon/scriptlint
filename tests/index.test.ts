import scriptlint from "../src/module";

jest.mock("fs");
jest.mock("path");

describe("index.ts", () => {
	it("should export a function", () => {
		expect(typeof scriptlint).toBe("function");
	});

	it("should throw with too much config", () => {
		expect(() => {
			scriptlint({
				packageFile: "foo/bar/baz",
				packageScripts: {
					foo: "bar"
				}
			});
		}).toThrowError(/not both/);
	});

	it("should throw with too little config", () => {
		expect(() => {
			scriptlint({});
		}).toThrowError(/location or a scripts/);
	});

	it("should lint files", () => {
		expect(
			scriptlint({
				packageFile: "real/existing/path/package.json"
			})
		).toMatchSnapshot();
	});

	it("should lint correctly", () => {
		expect(
			scriptlint({
				strict: true,
				packageScripts: {}
			}).issues.length
		).toBe(3);
	});

	it("should lint correctly", () => {
		expect(
			scriptlint({
				strict: false,
				packageScripts: {
					dev: "echo 1",
					test: "echo 1",
					start: "echo 1"
				}
			}).issues.length
		).toBe(0);
	});

	it("should fix correctly", () => {
		expect(
			Object.keys(
				scriptlint({
					fix: true,
					strict: true,
					packageScripts: {
						start: "echo 1",
						test: "echo 1",
						dev: "echo 1",
						foo: "bar"
					}
				}).scripts
			)
		).toEqual(["dev", "other:foo", "start", "test"]);
	});
});
