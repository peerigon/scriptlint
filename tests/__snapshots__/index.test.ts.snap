// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.ts should lint files 1`] = `
Object {
  "issues": Array [
    Object {
      "affected": undefined,
      "message": "must contain a \\"test\\" script (mandatory-test)",
      "name": "mandatory-test",
      "type": "warning",
    },
    Object {
      "affected": undefined,
      "message": "must contain a \\"start\\" script (mandatory-start)",
      "name": "mandatory-start",
      "type": "warning",
    },
    Object {
      "affected": undefined,
      "message": "must contain a \\"dev\\" script (mandatory-dev)",
      "name": "mandatory-dev",
      "type": "warning",
    },
  ],
  "scripts": Object {
    "foo": "bar",
  },
}
`;
